Common/Securities/Futures/FuturesExpiryFunctions.cs

```diff
@@ -40,44 +39,487 @@ public class FuturesExpiryFunctions
+                {
+                    businessDays--;
+                }
+                if (businessDays > 0) totalDays++;
+            } while (businessDays > 0);
+
+            return lastDayOfMonth.AddDays(-totalDays);
+        }
+
+        /// <summary>
+        /// Method to retrieve 9:30 a.m. Eastern Time (ET) on the 3rd Friday of the contract month
+        /// </summary>
+        private static DateTime ThirdFridayAtNineThirty(DateTime time)
+        {
+            var daysInMonth = DateTime.DaysInMonth(time.Year, time.Month);
+            return (from day in Enumerable.Range(1, daysInMonth)
+                    where new DateTime(time.Year, time.Month, day).DayOfWeek == DayOfWeek.Friday
+                    select new DateTime(time.Year, time.Month, day, 13, 30, 0)).ElementAt(2);
+        }
+
+        ///<summary>
+        /// Method to retrieve the business day prior to the 15th calendar day of the contract month.
+        /// </summary>
+        private static DateTime BusinessDayBeforeFifteenth(DateTime time)
+        {
+            var fifteenthDayOfMonth = new DateTime(time.Year, time.Month, 15);
+            // Get the previous businessday recursively
+            var previousDay = fifteenthDayOfMonth.AddDays(-1);
+            while (!previousDay.IsCommonBusinessDay() || USHoliday.Dates.Contains(previousDay))
+            {
+                previousDay = previousDay.AddDays(-1);
+            }
+            return previousDay;
+        }
+
+        /// <summary>
+        /// Method to retrieve 9:16 a.m. Central Time (CT) on the second business day immediately preceding the third Wednesday of the contract month (usually Monday).
+        /// </summary>
+        private static DateTime SecondBusinessDayPrecedingThirdWednesdayAtNineSixteen(DateTime time)
+        {
+            // Get Third Wednesday
+            var daysInMonth = DateTime.DaysInMonth(time.Year, time.Month);
+            var ThirdWednesday = (from day in Enumerable.Range(1, daysInMonth)
+                                  where new DateTime(time.Year, time.Month, day).DayOfWeek == DayOfWeek.Wednesday
+                                  select new DateTime(time.Year, time.Month, day)).ElementAt(2);
+
+            // Now get the second business day preceding third Wednesday
+            var previousDay = NthBusinessDay(ThirdWednesday, -2);
+            return new DateTime(previousDay.Year, previousDay.Month, previousDay.Day, 14, 16, 0);
+        }
+
+        /// <summary>
+        /// Method to retrieve 9:16 a.m. Central Time (CT) on the business day immediately preceding the third Wednesday of the contract month (usually Tuesday).
+        /// </summary>
+        private static DateTime BusinessDayPrecedingThirdWednesdayAtNineSixteen(DateTime time)
+        {
+            // Get Third Wednesday
+            var daysInMonth = DateTime.DaysInMonth(time.Year, time.Month);
+            var ThirdWednesday = (from day in Enumerable.Range(1, daysInMonth)
```

Comment:
Local variables typically do not capitalize the first letter in the name in C#. So, `thirdWednesday`